be/src/runtime/routine_load/data_consumer_pool.cpp

```diff
@@ -67,12 +67,18 @@ Status DataConsumerPool::get_consumer_grp(StreamLoadContext* ctx,
                 "PAUSE: Currently only support consumer group for Kafka data source");
     }
     DCHECK(ctx->kafka_info);
+    DCHECK(ctx->kafka_info->begin_offset.size() > 0);
```

Comment:
`DCHECK` means it must `> 0`;
So there should not be `consumer_num < 1`.

Why the begin offset is empty? Is it normal?

Reply:
In the create statement example below, you can see 
The user didn't specify the custom partitions.
So that 1、Fe will get all kafka partitions from server in routine load's scheduler.
The user set the default offset by datetime.
So that 2、Fe will get kafka offset by time from server in routine load's scheduler.

When 1 is success, meanwhile 2 is failed, the progress of this routine load may not contains any partitions and offsets.
Nevertheless, since newCurrentKafkaPartition which is get by kafka server may be always equal to currentKafkaPartitions, the wrong progress will never be updated.

Reply:
I think we should let the task fail directly if `ctx->kafka_info->begin_offset.size() == 0`, so that it can be easier to find error.

Reply:
Ok. Now the task will fail in get_consumer_grp method if the size of begin_offset is equal to 0.